### PR TITLE
GG-29901 Implement readiness probe REST endpoint

### PR DESCRIPTION
**Pull request by akorensh**

### CODE REVIEW

File: modules/core/src/main/java/org/apache/ignite/internal/processors/rest/handlers/probe/GridProbeCommandHandler.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        assert SUPPORTED_COMMANDS.contains(req.command());
+
+        switch (req.command()) {
+            case PROBE: {
+                if (log.isDebugEnabled())
+                    log.debug("probe command handler invoked.");
+
+                while (IgnitionEx.grid(ctx.igniteInstanceName()) == null) {
+                    try {
+                        Thread.sleep(1000L);
```
[the remark this answers is not in the document]
Comment (akorensh) [reply modules/core/src/main/java/org/apache/ignite/internal/processors/rest/handlers/probe/GridProbeCommandHandler.java]: implemented -- see latest pull request.
<p> Using startLatch status to determine whether kernal has started. </p>

Test checks whether this works:
      <p>1. start the grid on a seperate thread w/a plugin that will keep it waiting, at a point after rest http processor is ready, until signaled to proceed. </p>
      <p>2. when the grid.start() has reached the plugin init method(rest http processor has started now), issue a rest command against the non-fully started kernal. </p>
      <p>3. validate that the probe cmd has returned the appropriate erroneous code and message. </p>

File: modules/rest-http/src/test/java/org/apache/ignite/internal/processors/rest/protocols/http/jetty/GridProbeCommandTest.java [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+
+        Map<String, Object> probeRestCommandResponse;
+
+        probeRestCommandResponse = exectuteProbeRestRequest();
+
+        assertTrue(probeRestCommandResponse.get("response").equals("grid has started"));
+        assertEquals(0, probeRestCommandResponse.get("successStatus"));
+    }
+
+    public static Map<String, Object> exectuteProbeRestRequest() throws IOException {
```
Comment (slukyano) [inline modules/rest-http/src/test/java/org/apache/ignite/internal/processors/rest/protocols/http/jetty/GridProbeCommandTest.java:156]:
```suggestion
    public static Map<String, Object> executeProbeRestRequest() throws IOException {
```

**Review by slukyano — CHANGES_REQUESTED** [review]
Please fix the one typo highlighted and proceed with merge

File: modules/core/src/main/java/org/apache/ignite/internal/processors/rest/GridRestResponse.java [hunk 3]
```diff
@@ -50,6 +50,9 @@
     /** Success status. */
     private int successStatus = STATUS_SUCCESS;
 
+    //** HTTP REQUEST not allowed */
```
Comment (slukyano) [inline modules/core/src/main/java/org/apache/ignite/internal/processors/rest/GridRestResponse.java:53]:
```suggestion
    /** HTTP REQUEST not allowed */
```

File: modules/core/src/main/java/org/apache/ignite/internal/processors/rest/GridRestResponse.java [hunk 4]
```diff
@@ -50,6 +50,9 @@
...
+    public static final int SC_FORBIDDEN = 403;
```
Comment (slukyano) [inline modules/core/src/main/java/org/apache/ignite/internal/processors/rest/GridRestResponse.java:54]: better to use 503 "SERVICE UNAVAILABLE"

**Review by slukyano — CHANGES_REQUESTED** [review]

File: modules/core/src/main/java/org/apache/ignite/internal/processors/rest/GridRestResponse.java (hunk 4 above)
Comment (akorensh) [reply modules/core/src/main/java/org/apache/ignite/internal/processors/rest/GridRestResponse.java]: done.

**Pull request merged**